certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/unit/service/GetUnitMessagesDomainService.java

```diff
@@ -0,0 +1,51 @@
+package se.inera.intyg.certificateservice.domain.unit.service;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import se.inera.intyg.certificateservice.domain.action.model.ActionEvaluation;
+import se.inera.intyg.certificateservice.domain.action.model.CertificateActionType;
+import se.inera.intyg.certificateservice.domain.certificate.repository.CertificateRepository;
+import se.inera.intyg.certificateservice.domain.common.model.MessagesRequest;
+import se.inera.intyg.certificateservice.domain.common.model.MessagesResponse;
+import se.inera.intyg.certificateservice.domain.message.model.Message;
+import se.inera.intyg.certificateservice.domain.message.model.MessageType;
+import se.inera.intyg.certificateservice.domain.message.repository.MessageRepository;
+
+@RequiredArgsConstructor
+public class GetUnitMessagesDomainService {
+
+  private final MessageRepository messageRepository;
+  private final CertificateRepository certificateRepository;
+
+  public MessagesResponse get(MessagesRequest request, ActionEvaluation actionEvaluation) {
+    final var messages = messageRepository.findByMessagesRequest(request)
+        .stream()
+        .filter(message -> !isReminderOrAnswer(message))
+        .toList();
+
+    final var certificates = messages.stream()
+        .map(Message::certificateId)
+        .distinct()
+        .map(certificateRepository::getById)
+        .filter(certificate -> certificate.allowTo(
+                CertificateActionType.READ,
+                Optional.of(actionEvaluation)
+            )
+        ).toList();
+
+    final var filteredMessagesBasedOnActionEvaluation = messages.stream()
+        .filter(message -> certificates.stream()
+            .anyMatch(certificate -> certificate.id().id().equals(message.certificateId().id()))
+        ).toList();
+
+    return MessagesResponse.builder()
+        .messages(filteredMessagesBasedOnActionEvaluation)
+        .certificates(certificates)
+        .build();
+  }
+
+  private static boolean isReminderOrAnswer(Message message) {
+    return message.type() == MessageType.ANSWER || message.type() == MessageType.REMINDER;
```

Comment:
In this PR: https://github.com/sklintyg/core/pull/215/files#diff-3bf51e453e5aa08cb8e1d5c1cffe8ba0eae3e72dca3514e20f6d7006e9515cde we introduce a third type that should be ignored, called MISSING. Not sure how this PR should handle that, but im mentioning it so that we are aware of this and handle it. Maybe depending on who merges first? :)

Reply:
It's better if it's in the PR where the enum is introduced. If it's merged first I'll update this.